tools/depends/configure.ac

```diff
@@ -667,9 +667,9 @@ fi
 
 CXXFLAGS="$platform_cxxflags $platform_includes"
 CXX_CACHED="$CXX"
-AX_CXX_COMPILE_STDCXX([17], [noext], [mandatory])
+AX_CXX_COMPILE_STDCXX([20], [noext], [mandatory])
 CXX="$CXX_CACHED"
-platform_cxxflags="${platform_cxxflags} -std=c++17"
+platform_cxxflags="${platform_cxxflags} -std=c++20"
```

Comment:
we could keep 17 for depends for now, should get rid of the build issues

Reply:
This breaks `smctemp` on macOS. We override `CXXFLAGS` inside our makefile [here](https://github.com/xbmc/xbmc/blob/master/tools/depends/target/smctemp/Makefile#L19) but the native makefile needs `-std=c++17` to be [present](https://github.com/narugit/smctemp/blob/0.2.1/Makefile#L2C13-L2C49). I guess we need to override `CXXFLAGS` because of the different architectures but we should probably have `-Wall -std=c++17 -g -framework IOKit` as default `CXXFLAGS` in our makefile to have the same flags as upstream.

Reply:
thanks for finding this, let's see what happens